libsupport/include/katana/Result.h

```diff
@@ -229,6 +230,11 @@ class KATANA_EXPORT [[nodiscard]] ErrorInfo {
 
   ErrorInfo() : ErrorInfo(std::error_code()) {}
 
+  ErrorInfo(ErrorInfo &&) = default;
+  ErrorInfo& operator=(ErrorInfo&&) = default;
+  ErrorInfo(const ErrorInfo&) = default;
+  ErrorInfo& operator=(const ErrorInfo&) = default;
```

Comment:
I think copy constructors were intentionally omitted. This is because the `Context*` in `context_` points to a thread local. @ddn0 is this safe?

(The move should be "safe" I think. As "safe" as move ever is.)

Reply:
The copy constructors were actually already used in the current code. It doesn't compile if I `delete` them or define just the move constructor and assignment operator.

Reply:
These changes are benign though confusing. I don't see how they change much. The special member functions should be implicitly declared even without these explicit defaultings because `ErrorInfo` doesn't not declare its own special member functions.

Is `nvcc` confused by the `ErrorEnum` constructor below?

(And if we are going to default them, we might as well default them all including the destructor)

Reply:
I tried these because Boost docs said something about copy/move constructors/assignment operators. But these did not fix the problem, and it should be safe to remove them.